Extension/src/LanguageServer/configurations.ts

```diff
@@ -627,6 +696,10 @@ export class CppProperties {
             const configuration: Configuration = this.configurationJson.configurations[i];
 
             configuration.includePath = this.updateConfigurationStringArray(configuration.includePath, settings.defaultIncludePath, env);
+            if (settings.addNodeAddonIncludePaths) {
```

Comment:
configuration.includePath is checked/modified in the code below under certain settings. The "unmodified" configuration.includePath should be saved to some variable here to be used later.

Reply:
```
if (!configuration.includePath && !!this.defaultIncludes) {
                            configuration.includePath = this.defaultIncludes;
                        }
```
should check the unmodified/original variable and then do something like
```
                const includePath: string[] = configuration.includePath || [];
                configuration.includePath = includePath.concat(this.defaultIncludes);
```

Reply:
thanks for this; i feel like a dog watching tv when i work in this codebase.